### PR TITLE
Feat/select star

### PR DESCRIPTION
**Pull request by wilbertharriman**

### CODE REVIEW

**Review by SLMT — CHANGES_REQUESTED** [review]
You can try to make `*` viewed as one of fields in `Parser` and then replace it with all fields in `Verifier`.

**Comment by wilbertharriman** [comment]
1. Added suggested changes in `Parser` and `Verifier`
2. Use `ServerInit.loadTestbed()` for SELECT * integration test
3. Remove macOS from github actions. Otherwise, the limit on running github actions could be reached easily. When the limit is reached, github runners will not start and CI actions will fail after 6 hours.

File: src/main/java/org/vanilladb/core/query/planner/Verifier.java [hunk 1]
```diff
@@ -66,20 +56,35 @@ public static void verifyQueryData(QueryData data, Transaction tx) {
 			}
 		}
 
+		boolean allFields = false;
+
 		// examine the projecting field name
 		for (String fldName : data.projectFields()) {
+			if (fldName.equals("*")) {
```
Comment (SLMT) [inline src/main/java/org/vanilladb/core/query/planner/Verifier.java:63]: I think there should also be a constant for the wildcard in this class like how we did in `Lexer`.

**Review by SLMT — CHANGES_REQUESTED** [review]
Great improvement! `SELECT * ...` is an important syntax sugar for DB users. I think it is good to go once some my tiny requests are addressed.

File: src/main/java/org/vanilladb/core/query/parse/Lexer.java [hunk 2]
```diff
@@ -25,19 +25,22 @@
  * The lexical analyzer.
  */
 public class Lexer {
+	public static final String WILDCARD = "*";
```
Comment (wilbertharriman) [inline src/main/java/org/vanilladb/core/query/parse/Lexer.java:28]: WILDCARD is now a static and final field

File: src/main/java/org/vanilladb/core/query/planner/Verifier.java [hunk 3]
```diff
@@ -40,10 +30,12 @@
 import org.vanilladb.core.storage.metadata.TableMgr;
 import org.vanilladb.core.storage.tx.Transaction;
 
+import static org.vanilladb.core.query.parse.Lexer.WILDCARD;
```
Comment (wilbertharriman) [inline src/main/java/org/vanilladb/core/query/planner/Verifier.java:33]: import WILDCARD from `Lexer`

**Review by SLMT — APPROVED** [review]
Good to go!

**Pull request merged**